### PR TITLE
Add IR instructions documents

### PR DESCRIPTION
**Pull request by splasky**
Document is important.Add a document.

### CODE REVIEW

**Comment by jserv** [comment]
@splasky, Please follow the advice from [How to Write a Git Commit Message](https://chris.beams.io/posts/git-commit/) to amend your Git commits.
* Do not end the subject line with a period
* Use the body to explain what and why vs. how

File: Document/IR.md [hunk 1]
```diff
@@ -0,0 +1,105 @@
+# How IR work inside amacc
```
Comment (jserv) [inline Document/IR.md]: Use consistent naming `AMaCC`.

Comment (jserv) [reply Document/IR.md]: Change toplevel directory to `docs` instead of `Document`.

File: Document/IR.md [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+
+# Instructsion sets
+
+these instructions is for general purpose
+amacc opcode|       format      |       arm instructions        |comments
```
Comment (jserv) [inline Document/IR.md]: Simpy call it "opcode".

File: Document/IR.md [hunk 3]
```diff
@@ -0,0 +1,105 @@
+# How IR work inside amacc
+
+# Instructsion sets
+
+these instructions is for general purpose
```
Comment (jserv) [inline Document/IR.md]: Before explaining ISA, you should mention the design of stack-based IR. You can read further descriptions via https://github.com/lotabout/write-a-C-interpreter/tree/master/tutorial/en

File: Document/IR.md [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+
+# Arithmetic instructions
+
+Each operator has two arguments, the first one is store on the top of the stack,the second is stored in general register.After the calculation is done, the argument on the stack will be poped out and the result will be stored in general register. So you are not able to fetch the first argument from the stack after the calculation.
+
+You can see the above example to know how arithmetic instructions work.
+
+# compare instructions
+
+The `BZ` and `BNZ` instructions must be used with arithmetic instructions `EQ NE LT GT LE GE`
```
Comment (jserv) [inline Document/IR.md]: Change `EQ NE LT GT LE GE` to `EQ`, `NE`, etc.

**Comment by splasky** [comment]
Update the document. Code review is needed.

File: doc/IR.md [hunk 5]
```diff
@@ -0,0 +1,157 @@
+AMaCC Quick start
+===
+# How IR work inside AMaCC
+AMaCC use the stack based virtual machine for process the execute command。When the source file load into AMaCC,it will be translated to token，these token will be translated to AMaCC's specific IR，then use these IR to generated executed binary.
```
Comment (jserv) [inline doc/IR.md]: Fix the grammar.

**Pull request merged**